webui/react/src/components/CreateUserModal.tsx

```diff
@@ -194,13 +195,24 @@ const CreateUserModalComponent: React.FC<Props> = ({
       <Spinner
         spinning={user !== undefined && userRoles?.isNotLoaded && rbacEnabled && canAssignRoles({})}
         tip="Loading roles...">
-        <Form className={css.createUserModalForm} form={form} id={idPrefix + FORM_ID}>
+        <Form
+          className={css.createUserModalForm}
+          form={form}
+          id={idPrefix + FORM_ID}
+          onFieldsChange={() =>
+            setCanSubmit(
+              !hasErrors(form) &&
+                form.getFieldValue(USER_NAME_NAME) &&
+                (user ||
+                  (form.getFieldValue(USER_PASSWORD_NAME) &&
+                    form.getFieldValue(USER_PASSWORD_CONFIRM_NAME))),
+            )
```

Comment:
did you try this? https://ant.design/components/form#components-form-demo-validate-trigger
this looks a bit redudant

Reply:
I gave it a try but it didn't disable the button when fields were blank. I may have set it up incorrectly.

I like the idea of adding a switch if we're editing the user. At this point, the behaviors (and validation logic) of "create a user" and "edit a user" are pretty different -- would it make sense to just separate them as part of this ticket? It might result in two files that have very similar-looking structure; I'm not sure if that helps or hurts maintainability in the long run.